Assign to multitple windows, unbind & selfdestroy tooltip

LabAsim opened the pull request:
I have updated the Tooltip class to allow a assignment to either a tk.Widget/TK/Toplevel or a list of them (in order to assign the tooltip to multiple windows). 

I have also created functions to allow Tooltip to unbind from the previous tkinter Events and self destroy. 

I commited an example with 3 windows to show the use case.
@gnikit

gnikit reviewed (changes requested):
I am not entirely sure the problem this address and whether the proposed solution is the way to go. 

The unbind method is just a copy of the `create_binds`, surely it should be using `.unbind(...)`. Moreover, unbinding should occur in a destructor. The example and the proposed changes carry a lot of duplication, in terms of defining the bindings.
Finally you are mixing multiple documentation styles.

Also, please undo the unrelated changes to the contributing and gitignore.

This needs more careful consideration and a narrower scope in terms of proposed changes.

LabAsim commented:
The tooltip, in its current form, it is able to have bindings only to **one** widget. If someone wants to add the same tooltip to more windows (for example `tk.Tk` and a `Toplevel` window), he/she has to create the same tooltip for each window separately.

In my case, i wanted to use the tooltip as a loading indicator for the user in my tkinter app, so i needed to bind multiple windows (root and toplevels) to the same tooltip (with the loading message), run the web scraping methods and then unbind & destroy the tooltip. I did not managed to do it without passing a list to the `Tooltip`, This is functionality i tried to replicate in my commited example.  If this is not clear or there is a better way without modifying the existing Tooltip code, i would be glad to hear it.

As far the unbind method is concerned, you are right. It's the same function, it's my mistake commiting it. I have corrected it. Now it binds to an empty string, effectively replacing the previous bindings to Tooltips methods. (As it is also proposed [here](https://stackoverflow.com/a/23921363).)

I don't know what you mean by 
>  unbinding should occur in a destructor

Could you elaborate? Is not the unbinding & self_destroy method of the Tooltip sufficient?

I have undone the changes in contributing.md and gitignore and changed the docstrings according to the style of the docstring in the `Tooltip`. 

I think these changes address the latest [issue](https://github.com/gnikit/tkinter-tooltip/issues/39), too.

Let me know if anything is missing or needs to be changed/corrected.

gnikit commented:
This change is not a good fit for the project. Handling lists of widgets, certainly should not be part of the Tooltip itself, maybe you can use inheritance and make a specialised version for your project. 

With regards to the destroy function, it has been implemented in https://github.com/gnikit/tkinter-tooltip/pull/84.

I will be closing this PR now.
